src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/BookingTest.kt

```diff
@@ -914,6 +916,97 @@ class BookingTest : IntegrationTestBase() {
           .jsonPath("$.bed.id").isEqualTo(bed.id.toString())
           .jsonPath("$.bed.name").isEqualTo("test-bed")
           .jsonPath("$.assessmentId").isEqualTo("${assessment.id}")
+
+        // Assert the assessment is closed
+        val temporaryAccommodationAssessmentEntity = temporaryAccommodationAssessmentRepository.findByIdOrNull(assessment.id)
+        assertThat(temporaryAccommodationAssessmentEntity!!.completedAt).isNotNull()
+      }
+    }
+  }
+
+  @Test
+  fun `Create Temporary Accommodation Booking returns OK with correct body even closing assessment is failed for schema validation`() {
+    `Given a User`(roles = listOf(UserRole.CAS3_ASSESSOR)) { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, inmateDetails ->
+        val premises = temporaryAccommodationPremisesEntityFactory.produceAndPersist {
+          withYieldedLocalAuthorityArea { localAuthorityEntityFactory.produceAndPersist() }
+          withYieldedProbationRegion {
+            probationRegionEntityFactory.produceAndPersist { withYieldedApArea { apAreaEntityFactory.produceAndPersist() } }
+          }
+        }
+
+        val bed = bedEntityFactory.produceAndPersist {
+          withName("test-bed")
+          withYieldedRoom {
+            roomEntityFactory.produceAndPersist {
+              withName("test-room")
+              withYieldedPremises { premises }
+            }
+          }
+        }
+
+        val applicationSchema = temporaryAccommodationApplicationJsonSchemaEntityFactory.produceAndPersist {
+          withPermissiveSchema()
+        }
+
+        val application = temporaryAccommodationApplicationEntityFactory.produceAndPersist {
+          withCreatedByUser(userEntity)
+          withCrn(offenderDetails.otherIds.crn)
+          withProbationRegion(userEntity.probationRegion)
+          withApplicationSchema(applicationSchema)
+        }
+
+        val assessmentSchema = temporaryAccommodationAssessmentJsonSchemaEntityFactory.produceAndPersist {
```

Comment:
for my understanding - i guess this schema is invalid which leads to closure failing. If that's correct could the variable please be called 'invalidAssessmentSchema'?

Reply:
Its a valid schema only. But in the service layer we check the assessment.schema_id is equal to the as latest schema definition. The Date is used to find the latest schema id, if latest schema is not same as the assessment.schema_id then we get validation error. 

So inorder to test the validation error, if i create the schema without 'addedDate' then the schema-id comparison fails at service layer.